fabric/patches/2_0003-Deploy-qscc-via-opt_scc-framework.patch

```diff
@@ -0,0 +1,77 @@
+From cf6f634a9e15044ad3841c0ab35bd2156fa9f391 Mon Sep 17 00:00:00 2001
+From: michael steiner <michael.steiner@intel.com>
+Date: Wed, 19 Aug 2020 15:40:39 -0700
+Subject: [PATCH 3/3] Deploy qscc via opt_scc framework
+
+Signed-off-by: michael steiner <michael.steiner@intel.com>
+---
+ core/scc/qscc/query.go      | 11 +++++++++++
+ internal/peer/node/start.go |  5 +----
+ 2 files changed, 12 insertions(+), 4 deletions(-)
+
+diff --git a/core/scc/qscc/query.go b/core/scc/qscc/query.go
+index 82755e570..e1896182f 100644
+--- a/core/scc/qscc/query.go
++++ b/core/scc/qscc/query.go
+@@ -15,6 +15,9 @@ import (
+ 	"github.com/hyperledger/fabric/common/flogging"
+ 	"github.com/hyperledger/fabric/core/aclmgmt"
+ 	"github.com/hyperledger/fabric/core/ledger"
++	"github.com/hyperledger/fabric/core/peer"
++	"github.com/hyperledger/fabric/core/scc"
++	"github.com/hyperledger/fabric/core/scc/opt_scc"
+ 	"github.com/hyperledger/fabric/protoutil"
+ )
+ 
+@@ -226,3 +229,11 @@ func getBlockByTxID(vledger ledger.PeerLedger, rawTxID []byte) pb.Response {
+ func getACLResource(fname string) string {
+ 	return "qscc/" + fname
+ }
++
++func init() {
```

Comment:
I would add a comment here, quickly explaining that qscc is instantiated as an optional system chaincode using the opt factory. Just as a comment for us, I doubt that this go upstream but definitely shows how the factory can also used for non-fpc components.

Reply:
if you look at the patch it's obvious and there are also comments and it refers to the opt_scc package, really don't see what more to add ...

Reply:
well, I was thinking about having somewhere some background story why qscc now has a `init` but maybe there is a better place. Maybe this will be part of the Fabric PR.

Reply:
The init in query.go is now moot. Thinking a bit more on what i said below that we shouldn't use FPC as use-case, i realized that we should have a build-tag enabled version, so finally i did add now a mock scc to `opt_scc/mock` and then changed the qscc patch to have the same pattern: a file in `opt_scc` which just does the init, potentially build-tag enabled, and the actually scc code is elsewhere: so `query.go` is not touched anymore and the context is even more obvious now ...